src/serializer/functions.js

```diff
@@ -135,22 +146,27 @@ export class Functions {
     applyOptimizedReactComponents(this.realm, this.writeEffects, environmentRecordIdAfterGlobalCode);
   }
 
-  _callOfFunction(funcValue: FunctionValue): void => Value {
+  _callOfFunction(funcValue: FunctionValue, argModelString: void | StringValue): void => Value {
     let call = funcValue.$Call;
     invariant(call);
     let numArgs = funcValue.getLength();
     let args = [];
+    let argModel = argModelString !== undefined ? (JSON.parse(argModelString.value): ArgModel) : undefined;
```

Comment:
If a malformed string is given by the user, Prepack will crash here.
"Preparse" the string already in __optimize and turn a `JSON.parse` exception into a user-level error or `CompilerDiagnostics`. It's okay to parse the string twice.